File: 0755-reach-a-number/0755-reach-a-number.py
class Solution(object):
    def reachNumber(self, target):
        """
        :type target: int
        :rtype: int
        """
        target = abs(target)  
    
        n = 0
        sum_n = 0
        
        while sum_n < target:
            n += 1
            sum_n += n
        
        diff = sum_n - target
        
        if diff % 2 == 0:
            return n
        
        while (sum_n - target) % 2 != 0:
            n += 1
            sum_n += n
        
        return n
        

File: 0755-reach-a-number/0755-reach-a-number.java
class Solution {
    public int reachNumber(int target) {
        target = Math.abs(target);

        int sum = 0;
        int moves = 0;

        while (sum < target) {
            moves += 1;
            sum += moves;
        }

        int difference = sum - target;

        if (difference % 2 == 1) {
            moves += (moves % 2 == 0) ? 1 : 2;
        }

        return moves;

    }
}